packages/terra-search-field/src/terra-dev-site/doc/example/SearchFieldProgrammaticSet.jsx

```diff
@@ -0,0 +1,85 @@
+import React from 'react';
+import classNames from 'classnames/bind';
+import Button from 'terra-button/lib/Button';
+import SearchField from '../../../SearchField';
+import styles from '../../test/search-field/examplestyles.scss';
+
+const cx = classNames.bind(styles);
+
+class SearchFieldProgrammaticSet extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      searchText: '',
+      message: '',
+      text: '',
+    };
+
+    this.handleSearch = this.handleSearch.bind(this);
+    this.handleInvalidSearch = this.handleInvalidSearch.bind(this);
+    this.setTextClick = this.setTextClick.bind(this);
+    this.handleChange = this.handleChange.bind(this);
+    this.handleInput = this.handleInput.bind(this);
+  }
+
+  setTextClick() {
+    if (this.searchInput) {
+      this.searchInput.value = 'IpsumLorem';
+      let customEvent;
+      if (typeof (Event) === 'function') {
+        customEvent = new Event('input', {
+          bubbles: true,
+          cancelable: false,
+        });
+        customEvent.simulated = true;
```

Comment:
what's the simulated for? I don't see that in the event api

Reply:
Actually I think I left that in on accident. When I was trying to get onChange to be simulated, it was supposed to let React know that it was a "simulated" onChange event and not do the checks that prevent onChange from being programmatically fired like this.

Reply:
Yep, totally left it in on accident, still got same behavior without it. Fixed here https://github.com/cerner/terra-core/pull/2800/commits/b86945e9e390e61f2836c59e2fe31688bc0cd737#diff-5b54d3bd01a4e6076cc428ee0d369f6cL35